stylix/mk-target.nix

```
@@ -0,0 +1,102 @@
+{
+  name,
+  humanName,
+  autoEnable ? true,
+  extraOptions ? { },
+  configElements ? [ ],
+  generalConfig ? null,
+}:
+{ config, lib, ... }:
+let
+  cfg = config.stylix.targets.${name};
+in
+{
+  options.stylix.targets.${name} = lib.attrsets.unionOfDisjoint {
+    enable = config.lib.stylix.mkEnableTarget humanName autoEnable;
+  } extraOptions;
+
+  config =
+    let
+      args =
+        fn:
+        let
+          args = lib.functionArgs fn;
+          invalidArgs = lib.filterAttrs (_: value: value) args;
+          invalidArgsString = builtins.concatStringsSep ", " (
```

Not critical, but sometimes it's nice to escape any weirdly formatted names:

```suggestion
          invalidArgsString = builtins.concatMapStringsSep ", " lib.strings.escapeNixIdentifier (
```